mavisp/modes.py

```diff
@@ -106,7 +117,35 @@ def parse_metadata(self, data_dir, system):
             out_metadata['gitbook_entry'] = metadata['gitbook_entry']
         else:
             out_metadata['gitbook_entry'] = ''
-
+    
+        try:
+            structure_source = metadata.get("structure_source", None)
+            if structure_source not in self.structure_sources:
+                mavisp_criticals.append(
+                    MAVISpCriticalError(f"Invalid structure_source '{structure_source}' in metadata file. "
+                                        f"Expected one of {list(STRUCTURE_SOURCES.keys())}")
+                )
+            out_metadata["structure_source"] = structure_source
+            out_metadata["structure_description"] = self.structure_sources.get(structure_source, None)
+    
+            linker_design = metadata.get("linker_design", False)
+            if not isinstance(linker_design, bool):
+                mavisp_criticals.append(
+                    MAVISpCriticalError(f"Invalid value for linker_design: {linker_design}. Must be TRUE or FALSE.")
+                )
+            out_metadata["linker_design"] = linker_design
+    
+            pdb_id = metadata.get("pdb_id", None)
+            if structure_source in ["PDB", "Mod"] and not pdb_id:
+                mavisp_criticals.append(
+                    MAVISpCriticalError(f"structure_source '{structure_source}' requires a 'pdb_id' field.")
```

Comment:
requires a non-empty `pdb_id` field

I'm also not sure what happens when the pdb field is there but empty, is it in an empty string in that case? You should check that as well if this is the case